internal/cephfs/volumeoptions.go

```diff
@@ -110,30 +111,35 @@ func extractMounter(dest *string, options map[string]string) error {
 	return nil
 }
 
-func getClusterInformation(options map[string]string) (string, string, string, error) {
+func getClusterInformation(options map[string]string) (util.ClusterInfo, error) {
+	var clusterData util.ClusterInfo
 	clusterID, ok := options["clusterID"]
 	if !ok {
 		err := fmt.Errorf("clusterID must be set")
-		return "", "", "", err
+		return util.ClusterInfo{}, err
 	}
 
 	if err := validateNonEmptyField(clusterID, "clusterID"); err != nil {
-		return "", "", "", err
+		return util.ClusterInfo{}, err
 	}
 
 	monitors, err := util.Mons(csiConfigFile, clusterID)
 	if err != nil {
 		err = errors.Wrapf(err, "failed to fetch monitor list using clusterID (%s)", clusterID)
-		return "", "", "", err
+		return util.ClusterInfo{}, err
 	}
 
 	subvolumeGroup, err := util.CephFSSubvolumeGroup(csiConfigFile, clusterID)
 	if err != nil {
 		err = errors.Wrapf(err, "failed to fetch subvolumegroup using clusterID (%s)", clusterID)
-		return "", "", "", err
+		return util.ClusterInfo{}, err
 	}
-
-	return clusterID, monitors, subvolumeGroup, err
+	clusterData = util.ClusterInfo{
```

Comment:
You are creating a new `util.ClusterInfo` here, so it is not needed to define it as a `var` in the beginning of the function (use `clusterData := util.ClusterInfo{` ,with the `:=` here).